rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,154 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+
+### 4.2.4 Transaction Range Query
+
+If the IBLT from a `TransactionSet` message could be deconstructed, the local node MAY also send a `TransactionRangeQuery` message.
+This is only needed if the `LC` value of the `TransactionSet` message is equal or larger than `PSIZE*ceil(LC_req/PSIZE)`. 
+This means the peer has additional transactions outside the IBLT range.
+The `TransactionRangeQuery` message contains a `begin` and `end` parameter indicating a request for transactions which LC value is between those parameters.
+The `begin` MUST be set to `PSIZE*ceil(LC_req/PSIZE)` and the end MUST be set to `LC`. Both are inclusive.
+The message MUST contain a `token` that is to be used in the response.
+
+### 4.2.5 Transaction List
+
+The `TransactionList` message contains a list of transactions and the `token` from either the `TransactionListQuery` or `TransactionRangeQuery`.
+Each transaction in the message MUST relate to the original request. Transactions that resulted from a `TransactionListQuery` MUST have been present in that message.
+Transactions that resulted from a `TransactionRangeQuery` MUST have an LC value that is within the requested range.
+If this is not true, the entire message MUST be ignored.
+A `TransactionList` message MAY be broken up into smaller messages, each message should confirm to these rules. Each part MUST also use the same `token`.
+All transactions in the `TransactionList` message MUST be sorted by LC value (lowest first).
+
+## X. Private Transactions
+
+## Appendix A: Design decisions
+
+### A.1 IBLT parameters
+
+For a small network, 1024 buckets and 6 hashes is quite heavy. 
+For a nation-wide network this would be reasonable.
+These parameters are chosen to reduce the chance of a collision: two different transactions that are hashed into the exact same buckets.
+If there is a collision in the _set difference_, it is not be possible to deconstruct an IBLT.
+In the protocol described here, this would mean that an entire range of transactions would have to be sent.
+When this happens randomly, it wouldn't be such a problem, but it's also an attack vector.
+An attacker could craft pairs of transactions that would trigger other nodes into requesting large amounts of transactions. 
+This type of attack is called the [birthday attack](https://en.wikipedia.org/wiki/Birthday_attack).
+The only way to protect against this type of attack is to make the set large enough that the cost to the attacker would become to big.
+For an IBLT with 64 buckets and 4 hashes, the change for a collision is 50% after adding ~5000 keys.
+This would be trivial for current computers.
+Expanding this to 1024 buckets and 6 hashes, reaching a 50% chance for a collision is only achieved after adding more than a billion keys.
+This is still not much, but the keys used for the collision have to be created by real signed transactions.
+Even on modern hardware, this could take an hour.
```

Comment:
Are we quantum-proof? :-p

Reply:
The idea is to make the cost high enough and the impact low enough. The network will also be able to sync those transactions in a second. The cost of using a quantum computer will be high enough for the next 10 years or so?